### PR TITLE
fix: lazy suggest buffer size

### PR DESCRIPTION
**Pull request by Gabriela-David**
# Jira

[SOL-1407](https://uipath.atlassian.net/browse/SOL-1407)

[SOL-1407]: https://uipath.atlassian.net/browse/SOL-1407?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by anbalase** [comment]
since this is an improvement, I recommend we only deliver to master

File: projects/angular/components/ui-suggest/src/ui-suggest.component.ts [hunk 1]
```diff
@@ -719,13 +723,6 @@ export class UiSuggestComponent extends UiSuggestMatFormFieldDirective
                 takeUntil(this._destroyed$),
             )
             .subscribe(start => {
-                if (this._isLazyMode
```
Comment (anbalase) [inline projects/angular/components/ui-suggest/src/ui-suggest.component.ts:722]: why was this removed?

**Comment by anbalase** [comment]
@gheorghitahurmuz can you also pls review? you have good knowledge of this strategy.

File: projects/angular/components/ui-suggest/src/ui-suggest.component.ts (hunk 1 above)
Comment (Gabriela-David) [reply projects/angular/components/ui-suggest/src/ui-suggest.component.ts:722]: `_loadMore` calls `rangeLoad` and with the way lazy suggest works now, `rangeLoad` would be called twice when it is only needed once

File: projects/angular/components/ui-suggest/src/ui-suggest.component.ts [hunk 2]
```diff
@@ -1752,16 +1746,16 @@ export class UiSuggestComponent extends UiSuggestMatFormFieldDirective
     }
 
     private _shouldAddLoadingElementInLazyMode(currentResponse: ISuggestValue[]) {
-        return this._isLazyMode && currentResponse.length >= this.displayCount;
+                return this._isLazyMode && currentResponse.length >= this.displayCount;
     }
 
     private _addLoadingElementInLazyMode() {
-        const loadingElement = generateLoadingInitialCollection(this.intl.loadingLabel, 1)[0];
+        const loadingElement = generateLoadingInitialCollection(this.intl.loadingLabel, this._fetchCount);
```
Comment (gheorghitahurmuz) [inline projects/angular/components/ui-suggest/src/ui-suggest.component.ts:1753]: I believe there are several unused loading elements. For instance, we generate 20, but receive only 0 or 1 element from the backend. As a result, the remaining elements (e.g., 19 out of 20) are removed, which will affect the scrollbar.

File: projects/angular/components/ui-suggest/src/ui-suggest.component.ts [hunk 3]
```diff
@@ -1752,16 +1746,16 @@ export class UiSuggestComponent extends UiSuggestMatFormFieldDirective
...
 
         if (this.isDown) {
-            this._items.push(loadingElement);
+            this._items = [...this._items, ...loadingElement];
```
Comment (gheorghitahurmuz) [inline projects/angular/components/ui-suggest/src/ui-suggest.component.ts:1756]: why not `this._items.push(...loadingElement)`?

File: projects/angular/components/ui-suggest/src/ui-suggest.component.ts [hunk 4]
```diff
@@ -1752,16 +1746,16 @@ export class UiSuggestComponent extends UiSuggestMatFormFieldDirective
...
         } else {
-            this._items.unshift(loadingElement);
+            this._items = [ ...loadingElement, ...this._items];
```
Comment (gheorghitahurmuz) [inline projects/angular/components/ui-suggest/src/ui-suggest.component.ts:1758]: same ^

File: projects/angular/components/ui-suggest/src/ui-suggest.component.ts [hunk 5]
```diff
@@ -1655,7 +1649,7 @@ export class UiSuggestComponent extends UiSuggestMatFormFieldDirective
                     ),
                 ),
                 finalize(
-                    () => resetUnloadedState(this._items, mappedStart, mappedEnd),
+                    () => resetUnloadedState(this._items, mappedStart, mappedEnd, this._isLazyMode),
```
Comment (gheorghitahurmuz) [inline projects/angular/components/ui-suggest/src/ui-suggest.component.ts:1652]: I think we can implement the condition here and remove the flag related to lazy loading

File: projects/angular/components/ui-suggest/src/ui-suggest.component.ts [hunk 6]
```diff
@@ -1629,14 +1625,12 @@ export class UiSuggestComponent extends UiSuggestMatFormFieldDirective
 
         this._lazyLoadLastArgument = newArguments;
 
-        const mappedStart = this.isDown ?
-            this._isLazyMode ? this._items.length - 1 : start :
+        const mappedStart = this.isDown ? start :
             this._isLazyMode ? -1 : this._items.length - end - 1;
```
Comment (gheorghitahurmuz) [inline projects/angular/components/ui-suggest/src/ui-suggest.component.ts:1629]: If you have the condition `isLazy ? 0 : mappedStart, isLazy ? items.length : mappedEnd` implemented below, then you should modify it to `const mappedStart = this.isDown ? start : this._items.length - end - 1` here
